### PR TITLE
Milestone 3: Game logic

### PR DESCRIPTION
**Pull request by Davidosky007**
## Tic Tac Toe Game Milestone 3: Game logic
In this milestone i
* added the different files responsible for the game logic
* grouped the different methods and class responsible for game functions 
* followed the DRY rule throughout.

### CODE REVIEW

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @Davidosky007**

Great job with the project so far, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] There are supposed to be no puts or gets outside of the main.rb file, please make the changes that your main file has the logic to run game and you lib files have the methods to store and save the logic.

- [ ] when the game begins it says press any key to "restart" the game. it should say start

- [ ] when a player enters an invalid position, the game should display a warning. right now the game goes on an infinite loop when I enter a letter or any number that's not from 1-9.


---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @Davidosky007 

You've worked hard on this project but there's still a little more to be done before this milestone can be approved.

## [REQUIRED]

- [ ] There should be no `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [ ] Your game asked me to enter any key but I entered several keys yet the game did not start. I had to press the `enter` key to start the game. Fix that.
- [ ] Your game allows the second player to select the position already selected by the first player.
image
- [ ] When the game is a draw, your game outputs **The game is over! try again you might be lucky next time**. Your game should mention to the players that it's a **draw*.*

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

File: bin/main.rb [hunk 1]
```diff
@@ -1,124 +1,124 @@
 #!/usr/bin/env ruby
+require_relative '../lib/player'
+require_relative '../lib/board'
+
+class Game
+  def display_board(board)
```
Comment (mcrd25) [inline bin/main.rb:6]: [OPTIONAL]
Your main.rb file has a lot of methods, Ideally, your executable script has little to no methods. 
Instead, your methods within your classes should be written in such a way that it returns values. In this case, having a method that returns the board to print.
Consider this for other methods in this file.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Class for Board
+class Board
+  attr_accessor :board
```
Comment (mcrd25) [inline lib/board.rb:3]: [OPTIONAL]
Using  a 'getter' method is a good way to avoid use of `attr_accessor`

**Review by mcrd25 — APPROVED** [review]
# Changes Approved ✔️ 

Hi @Davidosky007 👋 

You have a few quirks but you have done a good job and meet the requirements.

Before you merge however please remove the commented out line in your Gemfile.
Please also consider the OPTIONAL comments below.

- As mentioned before only when a person presses enter the game starts so you should change the instructions. Same for quitting.

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email, or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya Douglas](mayacdouglas.com)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**